File: app/backend/internal/model/room.go
package model

type JoinPayload struct {
	Member
}

type LeavePayload struct {
	UserId  string `msgpack:"userId"`
	Session string `msgpack:"session"`
}

type RoomInfoPayload struct {
	Room        string `msgpack:"room"`
	MemberCount int    `msgpack:"memberCount"`
	MaxMembers  int    `msgpack:"maxMembers"`
}

File: app/frontend/src/protocol/payload/room.ts
import type { Member } from "@/models/member";

export type JoinPayload = Member;

export interface LeavePayload {
    userId: string;
    session: string;
}

export interface MembersPayload {
    members: Member[];
}

export interface RoomInfoPayload {
    room: string;
    memberCount: number;
    maxMembers: number;
    // owner?: string;
}
